v63002/u_inref/gtm8694.csh

```diff
@@ -0,0 +1,171 @@
+$ydb_dist/mumps -run zsystemfn^gtm8694
+$ydb_dist/mumps -run pipefn^gtm8694
+$ydb_dist/mumps -run trigmodfn^gtm8694
+echo "# TESTING DIRECTMODE"
+$ydb_dist/mumps -dir
+echo "# TESTING DSE"
+$ydb_dist/dse
+echo ""
+rm $ydb_dist/restrict.txt
+cat > $ydb_dist/restrict.txt << EOF
+CENABLE
+EOF
+chmod -w  $ydb_dist/restrict.txt
+(expect -d $gtm_tst/$tst/u_inref/gtm8694.exp > expect.outx) >& xpect.dbg
+if ($status) then
+	echo "EXPECT FAILED"
+endif
+echo "# TESTING CENABLE"
+perl $gtm_tst/com/expectsanitize.pl expect.outx > expect_sanitized.outx
+cat expect_sanitized.outx |& $grep CTRL
+echo ""
+echo "# -----------------------------------------------------------------------------"
+echo ""
+echo "# TESTING EACH FUNCTION WHEN RESTRICT.TXT HAS NO READ OR WRITE PERMISSIONS"
+echo ""
+# Not testing cenable, since this should restrict direct mode, which is necessary for testing cenable
+mv $ydb_dist/restrict.txt $ydb_dist/restrict1.txt
+cat > $ydb_dist/restrict.txt << EOF
+DIRECT_MODE
+EOF
+chmod -r -w $ydb_dist/restrict.txt
+$gtm_tst/com/lsminusl.csh $ydb_dist/restrict.txt | $tst_awk '{print $1,$9}'
+$ydb_dist/mumps -run breakfn^gtm8694
+$ydb_dist/mumps -run zbreakfn^gtm8694
+$ydb_dist/mumps -run zcmdlnefn^gtm8694 "ZCMDLNE was not ignored"
+$ydb_dist/mumps -run zeditfn^gtm8694
+$ydb_dist/mumps -run zsystemfn^gtm8694
+$ydb_dist/mumps -run pipefn^gtm8694
+$ydb_dist/mumps -run trigmodfn^gtm8694
+echo "# TESTING DIRECTMODE"
+$ydb_dist/mumps -dir
+echo "# TESTING DSE"
+$ydb_dist/dse
+echo ""
+echo ""
+echo "# ----------------------------------------------------------------------------"
+echo ""
+echo "# TESTING EACH FUNCTION WHEN RESTRICT.TXT LISTS OUR GROUP NAME (with readonly permissions)"
+echo ""
+rm $ydb_dist/restrict.txt
+set groupname=`id -g -n -r`
+echo ""
+cat > $ydb_dist/restrict.txt << EOF
+BREAK:$groupname
+ZBREAK:$groupname
+ZCMDLINE:$groupname
+ZEDIT:$groupname
+ZSYSTEM:$groupname
+CENABLE:$groupname
+PIPE_OPEN:$groupname
+DIRECT_MODE:$groupname
+DSE:$groupname
+TRIGGER_MOD:$groupname
+EOF
+chmod -w $ydb_dist/restrict.txt
+$gtm_tst/com/lsminusl.csh $ydb_dist/restrict.txt | $tst_awk '{print $1,$9}'
+setenv EDITOR ""
+$ydb_dist/mumps -run breakfn^gtm8694
+$ydb_dist/mumps -run zbreakfn^gtm8694
+$ydb_dist/mumps -run zcmdlnefn^gtm8694 "ZCMDLNE was not ignored"
+$ydb_dist/mumps -run zeditfn^gtm8694
+$ydb_dist/mumps -run zsystemfn^gtm8694
+$ydb_dist/mumps -run pipefn^gtm8694
+$ydb_dist/mumps -run trigmodfn^gtm8694
+echo "# TESTING DIRECTMODE"
+$ydb_dist/mumps -dir
+echo "# TESTING DSE"
+$ydb_dist/dse
+echo ""
+(expect -d $gtm_tst/$tst/u_inref/gtm8694.exp > expect.outx) >& xpect.dbg
+if ($status) then
+	echo "EXPECT FAILED"
+endif
+echo "# TESTING CENABLE"
+perl $gtm_tst/com/expectsanitize.pl expect.outx > expect_sanitized.outx
+cat expect_sanitized.outx |& $grep CTRL
+
+echo ""
+echo "# -------------------------------------------------------------------------"
+echo ""
+echo "TESTING EACH FUNCTION WHEN RESTRICT.TXT HAS WRITE PERMISSIONS"
```

Comment:
Same comment as before. For this test, randomly remove each line in restrict.txt with 50% chance. The output should not be affected because write permissions to restrict.txt means no restrictions whatsoever.